### PR TITLE
Gh 435 tidy up maven dependencies

### PR DESCRIPTION
**Pull request by t616178**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -108,17 +121,82 @@
...
+                <artifactId>scalap</artifactId>
+                <version>${scala.minor.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.xerial.snappy</groupId>
+                <artifactId>snappy-java</artifactId>
+                <version>${snappy.version}</version>
             </dependency>
         </dependencies>
     </dependencyManagement>
```
Comment (p013570) [inline pom.xml]: As we are making use of the maven dependency management section, we can remove the dependency versions from the other poms that use any of these dependencies.

E.g in serialisation-library/pom.xml
```xml
<dependency>
    <groupId>com.clearspring.analytics</groupId>
    <artifactId>stream</artifactId>
</dependency>
```

**Comment by gaffer01** [comment]
Is there any particular reason for using version 2.2.0 of Hadoop? That's relatively old - sticking with something like 2.6.3 or 2.7.0 might be better?

**Comment by p013570** [comment]
Accumulo 1.7.2 depends on Hadoop 2.2.0 so it is probably best to stick with 2.2.0 until we update the version of Accumulo.

File: rest-api/pom.xml [hunk 2]
```diff
@@ -101,7 +101,7 @@
         <dependency>
             <groupId>com.fasterxml.jackson.datatype</groupId>
             <artifactId>jackson-datatype-json-org</artifactId>
-            <version>2.3.3</version>
+            <version>${jackson.version}</version>
```
Comment (p013570) [inline rest-api/pom.xml]: Can we not just remove these versions? As the version within the dependency management section in the parent pom should be automatically used.

Comment (t616178) [inline rest-api/pom.xml]: This only works if the exact dependency is listed in the dependency management section - I was aiming for a balance between moving some things to the dependency management section of the parent POM and not having to list every single dependency there.

**Comment by t616178** [comment]
The release notes for Accumulo 1.7.2 state that 2.2.0 or newer is required. In fact, they run the Accumulo integration tests on Hadoop 2.6.3 - it should be fine (and probably preferred) to stick with the latest Hadoop 2.x.x release.

**Comment by p013570** [comment]
Let's leave the hadoop version as it is for now and raise a separate issue to update it.

**Pull request merged**